aten/src/ATen/native/Copy.cpp

```diff
@@ -10,11 +10,27 @@ namespace {
 
 template <typename self_T, typename src_T>
 void _copy__cpu(at::Tensor& self, const at::Tensor& src) {
-  at::CPU_tensor_apply2<self_T, src_T>(
-      self, src, [](self_T& self_val, const src_T& src_val) {
-        self_val = static_cast<self_T>(
+  bool serial_path = false;
+#ifdef _OPENMP
+  if (!at::in_parallel_region()) {
+    at::CPU_tensor_parallel_apply2<self_T, src_T>(
+        self, src, [](self_T& self_val, const src_T& src_val) {
+          self_val = static_cast<self_T>(
             static_cast<at::native::inter_copy_type_t<self_T>>(src_val));
-      });
+          }, 20000);
```

Comment:
This looks like magic number to me, do you have tests to different values?

Reply:
This number just copy from https://github.com/pytorch/pytorch/blob/bf5f30f39bc80cdf7f1e0770a63189b48bd9209b/aten/src/ATen/native/cpu/CopyKernel.cpp#L14,  perharps we need test different value. Thanks!

Reply:
I run the bechmark https://github.com/pytorch/pytorch/pull/19345#issuecomment-484351923 with setting the number to 1000, so that all can be parallized, it find that 10000 seems a better choice. I don't know whether it is ok for other dimension tensor.

tensor size | 1k | 2k | 3k | 4k | 5k | 8k | 10k | 20k | 50k | 80k | 100k | 110k | 120k | 150k | 180k
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
before(us) | 0.203866959 | 0.309558 | 0.429263 | 0.542231 | 0.654798 | 0.997729 | 1.256771 | 2.807269 | 6.1467457 | 9.6201134 | 12.30072 | 13.307371 | 14.76788 | 18.03192 | 21.8382311
after(us) threshold=1k | 0.932435989 | 0.666432 | 0.669942 | 0.635257 | 0.648742 | 0.677323 | 0.695462 | 0.791249 | 0.9452581 | 1.1859798 | 1.317997 | 1.2939167 | 1.113415 | 1.225438 | 1.36340618
after(us) threshold=20k | 0.23 | 0.34 | 0.46 | 0.58 | 0.69 | 1.02 | 1.25 | 0.76 | 0.96 | 1.1221647 | 1.3604546 | 1.3692379 | 1.468453 | 1.668973 | 1.82489395